oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/logout/OidcLogoutSuccessHandler.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.client.oidc.logout;
+
+import java.nio.charset.StandardCharsets;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.security.web.authentication.logout.SimpleUrlLogoutSuccessHandler;
+import org.springframework.util.Assert;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * A logout success handler for coordinating logout with an OIDC OP through the user agent.
+ *
+ * @author Josh Cummings
+ * @since 5.2
+ * @see <a href="http://openid.net/specs/openid-connect-session-1_0.html#RPLogout">RP-Initiated Logout</a>
+ * @see {@link org.springframework.security.web.authentication.logout.LogoutSuccessHandler}
+ */
+public class OidcLogoutSuccessHandler extends SimpleUrlLogoutSuccessHandler {
```

Comment:
Agreed, I like your first suggestion.

Reply:
I'm wondering if `OidcProviderLogoutSuccessHandler` makes more sense...since we're logging out at the provider?

Reply:
I like the `ClientInitiated` part because it ties us back to the spec.

Also, when I read `OidcProvider`, I hear something that belongs on the provider side, e.g. something that is "provider-initiated".

Aren't we always logging out at the provider (eventually)? Seems like we differentiate by saying where it starts.